### ISSUE TITLE
Add a custom registry entry for modules

### ISSUE THREAD

**Issue by dweidner** [getkirby-plugins/modules-plugin#4]
I currently try to move frequently used blueprints, templates and snippets into separate micro-plugins. Some of those plugins would benefit from your plugin. The problem is that the plugin expects all of the modules to be present in a single folder (`site/modules`  by default). It is possible to customize the path via a configuration setting, but what I would need is a way to specify multiple directories for each plugin that provides a module.

I think it would be great addition to implement a custom `Modules Registry Entry`. This would allow other plugins to add new modules via the registry while users are still allowed to place their own modules into `site/modules` as  default fallback.

The Kirby registry expects registry entries to live in the namespace `Kirby\Registry`.  In order to allow other plugins to register new modules via `$kirby->set('module', 'gallery', __DIR__ . DS . 'modules' . DS . 'gallery'); ?>` we would need to:
- Implement a new registry entry for modules
- Add the registry entry to Kirby’s autoloader

```
// extension/registry/module.php

namespace Kirby\Registry;

class Module extends Entry {
  public function set($name, $path, $automatic = false) { /*...*/ }
  public function get($name = null) { /* ... */ }
}
```

```
// modules.php

// Define autoloader
load([
  // ...
  'kirby\\registry\\module' => __DIR__ . DS . 'extensions' . DS . 'registry' . DS . 'module.php',
]);

// ...
```

Finally we have to rewrite the `Modules::allModules` in `lib/modules.php` to include plugin modules via the newly added registry. I could create a pull request, if you think that would be a nice addition to the plugin.

**Comment by lukasbestle** [comment]
Great idea! I have added this to my todo list.

**Comment by dweidner** [comment]
I could provide a pull request. The only thing I need is your preferred folder structure for the new entry.

**Comment by lukasbestle** [comment]
To be honest, a PR is not really going to make this a faster process. Thanks for the offer though. :)

**Comment by dweidner** [comment]
Ok... don't know what you mean, but hey, less work for me :). I was already digging a bit deeper into a core. Maybe implementing a custom kirby component for modules would be awesome as well, as it might allow plugin authors to customize some of the behaviors of the module plugin (e.g. the validation process).

**Comment by lukasbestle** [comment]
The registry is now implemented in version 1.2.0 of the Modules plugin.

Regarding the component: I want to keep the plugin as simple as possible. Adding a component would make the plugin much harder to debug. If you need changed logic, it would be best to fork the plugin. Let me know what change exactly you need (in a separate issue), maybe there's already a solution.
